### PR TITLE
OOP mile stone two

### PR DESCRIPTION
**Pull request by rootDEV2990**
Can you please help us check our OOP mile stone two progress want to make sure we hit the requirements. Thanks =) 👋

Notes from Pathwrite

- Main.rb file should be responsible for user interaction. In this milestone you should only create content in this file.
Think about steps needed to play the game:

- what do you need from player?
- what do you need to display for player?

- Display prompts for users (hint: puts/print), collect user input (hint: get.chomp) and only mock the basic flow of the game (you can use puts or inline comments to say for example “Now, your move is displayed on the board”).
- Make sure that you can run your game with your executable file and play with the basic flow.
- The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. **It does not need to work yet - this will be done in the next milestone.**

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,2 +1,109 @@
...
+   if move?(index)
+    make_move(index, player)
+     display_board
+   else
+     turn
+   end
+end
+
+# returns the number of turns
+def palyer_count
```
Comment (mosesogwo) [inline bin/main]: check spelling - palyer

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @rootDEV2990, @Joecleverman :clap:. Good start with this milestone:+1:**

From your code, I can see that you have thought about the following:
:heavy_check_mark: inform players of whose turn it is
:heavy_check_mark: ask the player to select from available moves
:heavy_check_mark: check validity of player move
:heavy_check_mark: display board after each move

Here are the things you should take care of now:
:x: In this milestone, you should only create content in main.rb, but you already have content in _lib/board.rb_
:x: You don't have to group objects in classes in this milestone. Grouping objects in classes will be the goal of the next milestone.
:x: You need to show that you have thought about win and draws for the game. The game loop that terminates after nine times should be changed to terminate when there is a win or a draw instead.
:x: In _player\_move_, you are creating a new Board in each turn of play. Check this properly. You should hold your board in a variable since you are using a single board throughout the game play.
:x: The player turn doesn't change when I play with the flow. I know you are not expected to make everything work but from your code, I think you are expecting it to work.
:x: Check your indentation, you should use two spaces for indentation.
:x: Be careful with typos

Summarily, you should remove the classes for now and layout your components and the game flow only in _bin/main.rb_, taking into consideration the points noted above.


_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by rootDEV2990** [comment]
Fixed some of issues from last CR. Please help us review changes.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hi Team.👋

Good job on this @rootDEV2990  and @techbabouche 🚀

However, let's make this few corrections:

- [ ] The loop shouldn't break after 10 rounds. Think of a better condition to end this game perhaps when we find a winner or a tie and incorporate this.

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAVE A SAFE DAY👨‍💻

**Comment by mikenath223** [comment]
In addition, please fix the file conflicts on this branch.

**Comment by rootDEV2990** [comment]
Fixed conflicts with dev branch for CR.

**Review by serdg0 — APPROVED** [review]
### :fireworks: :fireworks: Congratulations! @rootDEV2990 @Joecleverman :fireworks: :fireworks: :fire:  

Go on and merge.

Keep the good work in your next milestone! 

**Happy Coding! :nerd_face: :computer:** 
_Sergio Diaz_

**Pull request merged**